docs/spec/spec.md

```diff
@@ -0,0 +1,238 @@
+# Specification: Ballerina NATS Library
+
+_Owners_: @aashikam @shafreenAnfar  
+_Reviewers_: @shafreenAnfar  
+_Created_: 2020/10/28 
+_Updated_: 2021/11/29  
+_Issue_: [#2214](https://github.com/ballerina-platform/ballerina-standard-library/issues/2214)
+
+# Introduction
+This is the specification for NATS standard library which is used to send and receive messages by connecting to the NATS server. 
+This library is programmed in the [Ballerina programming language](https://ballerina.io/), which is an open-source programming language for the cloud
+that makes it easier to use, combine, and create network services.
+
+# Contents
+
+1. [Overview](#1-overview)
+2. [Connection](#2-connection)
+3. [Publishing](#3-publishing)
+5. [Listening for Messages](#4-listening-for-messages)
```

Comment:
I think to be symmetric we can use `Subscribing`

Reply:
Also in each section it would be good to have the definition of relevant client and service objects.

Reply:
Fixed in 10fb3b2fa3dc16aaaae872decb02ac40382e2729